### PR TITLE
DAMIEN-287: Added stricter validation for selected eval periods

### PR DESCRIPTION
**Pull request by cesarvh**
https://jira-secure.berkeley.edu/browse/DAMIEN-287

### CODE REVIEW

File: src/components/evaluation/EvaluationTable.vue [hunk 1]
```diff
@@ -397,7 +397,11 @@ export default {
       return this.$currentUser.isAdmin || !this.isSelectedTermLocked
     },
     rowValid() {
-      return this.rules.currentTermDate(this.selectedStartDate) === true
+      const sisDates = this.evaluations
+        .map(e => e.id === this.editRowId ? [e.courseStartDate, e.courseEndDate] : '')
+        .filter(e => !!e)
```
Comment (pauline2k) [inline src/components/evaluation/EvaluationTable.vue]: @cesarvh For purposes of finding the object in the evaluations array matching `this.editRowId`, Lodash `find` might be more efficient and readable than mapping to a transient array: https://lodash.com/docs/4.17.15#find

File: src/components/evaluation/EvaluationTable.vue [hunk 2]
```diff
@@ -518,7 +522,7 @@ export default {
         this.evaluations.filter(ev => !ev.isSelected).forEach(ev => this.updateEvaluationsSelected(ev.id))
         this.selectedEvals = this.$_.map(this.evaluations, 'id')
       }
-    }
+    },
```
Comment (pauline2k) [inline src/components/evaluation/EvaluationTable.vue:571]: Extra comma might be confusing the JS parser?

File: src/components/evaluation/EvaluationTable.vue [hunk 3]
```diff
@@ -532,13 +536,29 @@ export default {
...
+          lastStartDate = courseLength < 90 ? paddedDate.subtract(13, 'day') : paddedDate.subtract(20, 'day')
+          
+        } else {
+          lastStartDate = courseLength < 90 ? lastStartDate.subtract(13, 'day') : lastStartDate.subtract(20, 'day')          
+        }
+        return formattedSelectedDate <= lastStartDate.format('YYYY-MM-DD') ? true : 'Date must be within current term.'
       }
-      return 'Date must be within current term.'
-    }
+    },
```
Comment (pauline2k) [inline src/components/evaluation/EvaluationTable.vue]: Extra comma, ditto

File: src/components/evaluation/EvaluationTable.vue [hunk 4]
```diff
@@ -532,13 +536,29 @@ export default {
...
+
+      const courseLength = this.$moment(formattedCourseEnd).diff(formattedCourseStart, 'days')
+      let lastStartDate =  this.$moment(sisEndDate)
+
+      if (this.selectedTerm.name.includes('Summer')) {
+          lastStartDate = courseLength < 90 ? lastStartDate.subtract(13, 'day') : lastStartDate.subtract(20, 'day')
+          return formattedSelectedDate <= lastStartDate.format('YYYY-MM-DD') ? true : 'Date must be within current term.'
+      } else {
+        
+        if (formattedCourseEnd <= this.$config.currentTermDates.end) {
```
Comment (pauline2k) [inline src/components/evaluation/EvaluationTable.vue]: @cesarvh This logic all looks correct to me, except that the value in `this.$config.currentTermDates.end` is just a loose value that we implemented for first-pass sanity checking and not the default meeting end date for the term. (See damien.lib.berkeley.current_term_dates, which is the server-side source for these values.) What we actually want to test against are the default meeting dates for the term as provided by damien.lib.queries.get_default_meeting_dates; you might need to add these values to the config feed.

**Comment by cesarvh** [comment]
Realized that I missed there already being `meetingDates.start` and `meetingDates.end` so I removed the `courseBeginDate` and `courseEndDate` so tweaking `config_controller.py` wasn't necessary

File: src/components/evaluation/EvaluationTable.vue [hunk 5]
```diff
@@ -532,16 +535,32 @@ export default {
...
```
Comment (pauline2k) [inline src/components/evaluation/EvaluationTable.vue]: @cesarvh Code in this method probably needs to change to reference the new `defaultTermDates`?

File: tests/api/test_config_controller.py [hunk 6]
```diff
@@ -69,6 +69,7 @@ def test_current_term(self, app, client, fake_auth):
         response = client.get('/api/config')
         assert response.status_code == 200
         data = response.json
+        print(data)
```
Comment (pauline2k) [inline tests/api/test_config_controller.py]: @cesarvh Stray `print` in the test

**Pull request merged**